baremetal/cluster_manager_test.go

```diff
@@ -0,0 +1,290 @@
+	}
+	if err := infrav1.AddToScheme(s); err != nil {
+		panic(err)
+	}
+	if err := corev1.AddToScheme(s); err != nil {
+		panic(err)
+	}
+	if err := bmh.SchemeBuilder.AddToScheme(s); err != nil {
+		panic(err)
+	}
+	return s
+}
+
+func newCluster(clusterName string) *clusterv1.Cluster {
+	return &clusterv1.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Cluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: namespaceName,
+		},
+		Spec: clusterv1.ClusterSpec{
+			InfrastructureRef: &v1.ObjectReference{
+				Name:       baremetalClusterName,
+				Namespace:  namespaceName,
+				Kind:       "InfrastructureConfig",
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1alpha2",
+			},
+		},
+		Status: clusterv1.ClusterStatus{
+			InfrastructureReady: true,
+		},
+	}
+}
+
+func newBareMetalCluster(baremetalName string, ownerRef *metav1.OwnerReference,
+	spec *infrav1.BareMetalClusterSpec, status *infrav1.BareMetalClusterStatus) *infrav1.BareMetalCluster {
+	if spec == nil {
+		spec = &infrav1.BareMetalClusterSpec{}
+	}
+	if status == nil {
+		status = &infrav1.BareMetalClusterStatus{}
+	}
+	ownerRefs := []metav1.OwnerReference{}
+	if ownerRef != nil {
+		ownerRefs = []metav1.OwnerReference{*ownerRef}
+	}
+
+	return &infrav1.BareMetalCluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "BareMetalCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            baremetalClusterName,
+			Namespace:       namespaceName,
+			OwnerReferences: ownerRefs,
+		},
+		Spec:   *spec,
+		Status: *status,
+	}
+}
+
+func newMachine(clusterName, machineName string, bareMetalMachineName string) *clusterv1.Machine {
+	machine := &clusterv1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Machine",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      machineName,
+			Namespace: namespaceName,
+			Labels: map[string]string{
+				clusterv1.MachineClusterLabelName: clusterName,
+			},
+		},
+	}
+	if bareMetalMachineName != "" {
+		machine.Spec.InfrastructureRef = v1.ObjectReference{
+			Name:       bareMetalMachineName,
+			Namespace:  namespaceName,
+			Kind:       "BareMetalMachine",
+			APIVersion: infrav1.GroupVersion.String(),
+		}
+	}
+	return machine
+}
+
+func newBareMetalMachine(name string, ownerRef *metav1.OwnerReference,
+	spec *infrav1.BareMetalMachineSpec, status *infrav1.BareMetalMachineStatus) *infrav1.BareMetalMachine {
+
+	if spec == nil {
+		spec = &infrav1.BareMetalMachineSpec{}
+	}
+	if status == nil {
+		status = &infrav1.BareMetalMachineStatus{}
+	}
+	ownerRefs := []metav1.OwnerReference{}
+	if ownerRef != nil {
+		ownerRefs = []metav1.OwnerReference{*ownerRef}
+	}
+
+	return &infrav1.BareMetalMachine{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "BareMetalMachine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name,
+			Namespace:       namespaceName,
+			OwnerReferences: ownerRefs,
+		},
+		Spec:   *spec,
+		Status: *status,
+	}
+}
```

Comment:
does this need to be redefined ? If it is the same functions as in the machine_manager_test.go, then it could be defined only once since it is the same package

Reply:
newMachine and newBareMetalMachine are "common" for machine mgr and cluster mgr but currently implementations have that much delta that I shall do the consolidation in a separate commit.
thus these two functions are renamed in cluster mgr as newMachineClMgr and newBareMetalMachineClMgr.